setup.cfg

```diff
@@ -124,8 +126,7 @@ filterwarnings =
     ignore:leap-second auto-update failed
     # See https://github.com/astropy/extension-helpers/issues/23
     ignore:Distutils was imported before Setuptools
-    # toolz internal deprecation warning https://github.com/pytoolz/toolz/issues/500
-    ignore:The toolz.compatibility module is no longer needed:DeprecationWarning
+    ignore:Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
```

Comment:
Why is this warning being thrown?

Reply:
I thought this warning was thrown when you set MPL to use agg. I have seen this locally at times with the figure tests.

Reply:
pytest-mpl dynamically switches the backend, but I don't understand how we avoided this in our own code before this PR. As I thought we set the backend to Agg anyway?

Reply:
IN tox we do set it to AGG and I have seen this warning but it is a good question on how have we avoided this before.

Reply:
we set Agg in the conftest.py as well: https://github.com/sunpy/sunpy/blob/master/sunpy/conftest.py#L17-L22

Reply:
Do we need both?

Reply:
:man_shrugging:

Reply:
What's the stack trace when this warning is thrown? Maybe pytest-mpl needs to be setting it to agg somewhere?

Reply:
I think this is being raised because pytest-mpl is setting it to agg and we are calling .show inside of our peek methods.

Reply:
I can't replicate this inside tox, but the tests should run outside it just as well. The full traceback is:

```
___________________________________________________________ test_heliographic_peek ____________________________________________________________

cls = <class 'matplotlib.backends.backend_agg._BackendAgg'>

    @classmethod
    def show(cls, *, block=None):
        """
        Show all figures.
    
        `show` blocks by calling `mainloop` if *block* is ``True``, or if it
        is ``None`` and we are neither in IPython's ``%pylab`` mode, nor in
        `interactive` mode.
        """
        managers = Gcf.get_all_fig_managers()
        if not managers:
            return
        for manager in managers:
            try:
>               manager.show()  # Emits a warning for non-interactive backend.

../../.virtualenvs/pytest-mpl/lib/python3.8/site-packages/matplotlib/backend_bases.py:3524: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <matplotlib.backend_bases.FigureManagerBase object at 0x7ff5ab60a1c0>

    def show(self):
        """
        For GUI backends, show the figure window and redraw.
        For non-GUI backends, raise an exception, unless running headless (i.e.
        on Linux with an unset DISPLAY); this exception is converted to a
        warning in `.Figure.show`.
        """
        # This should be overridden in GUI backends.
        if cbook._get_running_interactive_framework() != "headless":
>           raise NonGuiException(
                f"Matplotlib is currently using {get_backend()}, which is "
                f"a non-GUI backend, so cannot show the figure.")
E           matplotlib.backend_bases.NonGuiException: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.

../../.virtualenvs/pytest-mpl/lib/python3.8/site-packages/matplotlib/backend_bases.py:2728: NonGuiException

During handling of the above exception, another exception occurred:

heliographic_test_map = <sunpy.map.mapbase.GenericMap object at 0x7ff571b1b0a0>
SunPy Map
---------
Observatory:             
Instrument:              
Detector:               ...    -0.02115 ],
       [-0.030485, -0.030435, -0.030385, ..., -0.03059 , -0.030535,
        -0.030485]], dtype=float32)

    @figure_test
    def test_heliographic_peek(heliographic_test_map):
>       heliographic_test_map.peek()

sunpy/map/tests/test_plotting.py:179: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
sunpy/visualization/visualization.py:22: in show_figure
    plt.show()
../../.virtualenvs/pytest-mpl/lib/python3.8/site-packages/matplotlib/pyplot.py:353: in show
    return _backend_mod.show(*args, **kwargs)
../../.virtualenvs/pytest-mpl/lib/python3.8/site-packages/matplotlib/backend_bases.py:3526: in show
    cbook._warn_external(str(exc))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

message = 'Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.', category = None

    def _warn_external(message, category=None):
        """
        `warnings.warn` wrapper that sets *stacklevel* to "outside Matplotlib".
    
        The original emitter of the warning can be obtained by patching this
        function back to `warnings.warn`, i.e. ``cbook._warn_external =
        warnings.warn`` (or ``functools.partial(warnings.warn, stacklevel=2)``,
        etc.).
        """
        frame = sys._getframe()
        for stacklevel in itertools.count(1):  # lgtm[py/unused-loop-variable]
            if frame is None:
                # when called in embedded context may hit frame is None
                break
            if not re.match(r"\A(matplotlib|mpl_toolkits)(\Z|\.(?!tests\.))",
                            # Work around sphinx-gallery not setting __name__.
                            frame.f_globals.get("__name__", "")):
                break
            frame = frame.f_back
>       warnings.warn(message, category, stacklevel)
E       UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.

../../.virtualenvs/pytest-mpl/lib/python3.8/site-packages/matplotlib/cbook/__init__.py:2108: UserWarning
```

Reply:
🤦 of course this is exactly what is expected, silly me for not realising. Could you drop a comment in above the warning ignore along the lines of ".peek() calls plt.show(), that raises a warning in the test suite because we use the agg backend"